docker/images/pinot/etc/jmx_prometheus_javaagent/configs/broker.yml

```diff
@@ -1,76 +1,90 @@
 rules:
-- pattern: "\"org.apache.pinot.common.metrics\"<type=\"BrokerMetrics\", name=\"pinot.broker.([^\\.]*?).authorization\"><>(\\w+)"
-  name: "pinot_broker_authorization_$2"
+- pattern: "\"org.apache.pinot.common.metrics\"<type=\"BrokerMetrics\", name=\"pinot.broker.(([^\\.]+)\\.)?([^\\.]*?).authorization\"><>(\\w+)"
```

Comment:
Trying to understand this regex:
1. What does `*?` stands for?
2. Seems like we don't need `\\` before `.`

Reply:
Not really sure about `*?`.
`(([^\\.]+)\\.)?` is the new regex being added to match database prefix with `.` . I think `\\` is needed to escape `.` as that's a literal.

Reply:
`\\` is not needed to escape `.` (notice that we don't escape other `.`)
I think we can simplify the matching with `(([^.]+).)?([^.]+)`. Can you please try it and see if it works?

Reply:
Only the expression between `()` is considered as regex that's why we don’t escape the `.` outside the brackets.
Tried testing without escaping the `.` inside brackets and its behaving as a match all expression, hence we need to escape the `.` inside brackets.

Regarding `*?`, I can raise a separate PR to replace it with `+`.

Reply:
Interesting. Maybe Prometheus regex works differently from Java regex?

Reply:
My understanding is wrong mostly. The `.` requires escaping throughout I suppose. The only reason its working right now is because its surrounded by specific patterns, so the match all `.` is only matching actual `.` character here.
If you see the catch all rules, there we are escaping the `.` even outside the brackets.
We need to fix this. Will open a separate PR on this.

Reply:
@shounakmk219 I think your understanding about `.` is correct. We wanted to fix it, we didn't because the reason you mentioned, it just worked.

For `*?`, I don't know about promethues, but `*` vs `*?` is mainly about greedy and non greedy matching. See https://stackoverflow.com/questions/3075130/what-is-the-difference-between-and-regular-expressions

Reply:
Nice, then in our case existing non-greedy approach will be optimal as this is supposed to only match the table name which is usually not very long. So no change on `*?`.

Reply:
So here I can see 3 things we need to figure out:
1. Do we need to escape `.` when it is within `[]`? I tried it in java regex and `[^\\.]` and `[^.]` are equivalent.
2. Do we need to escape `.` when it is not in the matching parenthesis?
3. Do we need `*?` over `*`? IMO if `.` is handled correctly, greedy one (`*`) should be better

Reply:
Came across a case where the the unescapped `.` is breaking the metric parsing 
Metric endpoint result : 
```
pinot_controller_segmentCount_Value{database="ncludingReplaced",table="ncludingReplaced.test",tableType="OFFLINE",} 1.0
```
Actual metric : 
```
pinot.controller.segmentCountIncludingReplaced.test_OFFLINE
```
Rule it was supposed to match : 
```
pattern: "\"?org\\.apache\\.pinot\\.common\\.metrics\"?<type=\"?\\w+\"?, name=\"?pinot\\.(\\w+)\\.(\\w+)\\.((\\w+)\\.)?(\\w+)_(OFFLINE|REALTIME)\\\"?><>(\\w+)"
```
Rule it actually matched : 
```
pattern: "\"org.apache.pinot.common.metrics\"<type=\"ControllerMetrics\", name=\"pinot.controller.segmentCount.(([^\\.]+)\\.)?([^\\.]*?)_(OFFLINE|REALTIME)\"><>(\\w+)"
```

This answers your question @Jackie-Jiang , we need to escape `.` at all places in the pattern string to parse it as a dot literal.